compiler/modules/front.syntax/ast.bal

```diff
@@ -30,12 +30,10 @@ public type ImportDecl record {|
 |};
 
 public type FunctionDefn record {|
-    *PositionFields;
+    *FunctionTypeDesc;
     readonly string name;
     ModulePart part;
     Visibility vis;
-    FunctionTypeDesc typeDesc;
-    string[] paramNames;
```

Comment:
I think we should have

```
public type FunctionParam record {|
    *PositionFields;
    string name;
    Position namePos;
    TypeDesc td;
|};
```

and then a member

```
   FunctionParam[] params;
```

This params array would be `===` to typeDesc.params

Reply:
fixed